File: Java/app/src/main/java/com/divya/rememberyourpasswords/data/UserRepository.java
package com.divya.rememberyourpasswords.data;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.Transformations;

/**
 * Created by divya on 5/16/2018.
 */

public class UserRepository {

    private final UserAccountDao userAccountDao;
    private static UserRepository instance;
    private LiveData<UserAccount> userAccountLiveData;

    private UserRepository(UserAccountDao userAccountDao)
    {
        this.userAccountDao = userAccountDao;
    }

    public static UserRepository getInstance(UserAccountDao userAccountDao)
    {
        if(instance == null)
        {
            instance = new UserRepository(userAccountDao);
        }
        return instance;
    }

    public boolean isValidAccount(String username, final String password)
    {

        UserAccount userAccount = userAccountDao.getAccount(username);
        return userAccount.getPassword().equals(password);
    }

    public void insertUser(String username, String password)
    {
        UserAccount account = new UserAccount(username, password);
        userAccountDao.insert(account);
    }
}

File: Kotlin/app/src/main/java/com/divya/signup_login_kotlin/data/UserRepository.kt
package com.divya.rememberyourpasswords.data

import android.arch.lifecycle.LiveData
import android.arch.lifecycle.Transformations

/**
 * Created by divya on 5/16/2018.
 */

class UserRepository private constructor(private val userAccountDao: UserAccountDao) {
    private val userAccountLiveData: LiveData<UserAccount>? = null

    fun isValidAccount(username: String, password: String): Boolean {

        val userAccount = userAccountDao.getAccount(username)
        return userAccount.password == password
    }

    fun insertUser(username: String, password: String) {
        val account = UserAccount(username, password)
        userAccountDao.insert(account)
    }

    companion object {
        private var instance: UserRepository? = null

        fun getInstance(userAccountDao: UserAccountDao): UserRepository {
            if (instance == null) {
                instance = UserRepository(userAccountDao)
            }
            return instance!!
        }
    }
}
